### PR TITLE
Add py-wxflow to global-workflow-env

### LINKED ISSUE #1149: [INSTALL]: py-wxflow
### Package name

py-wxflow

### Package version/tag

0.2.0 (latest)

### Build options

n/a

### Installation timeframe

Include in unified-env for 1.8.0 release; for now, install under 1.7.0 in chained environment based on unified-env on Hera, Orion, Hercules, Jet, S4 and WCOSS2.

### Other information

_No response_

### WCOSS2

- [ ] Check this box if and only if your package should be installed on WCOSS2 Cactus and Dogwood (all spack-stack packages will be installed on Acorn). If not, you may disregard the rest of the items below and submit this request.

### WCOSS2: General questions

_No response_

### WCOSS2: Installation and testing

_No response_

### WCOSS2: Technical & security review list

- [ ] The code is mature, stable, and production ready
- [ ] The code is does not and cannot use the internet, and does not contain URLs (http, https, ftp, etc.) except in comments
- [ ] The package does not contain prebuilt binary files that have not been approved by NCO security review
- [ ] The code has no publicly disclosed cybersecurity vulnerabilities and exposures (search https://cve.mitre.org/cve/)
- [ ] The code is not prohibited by DHS, DOC, NOAA, or NWS
- [ ] The code comes from a trusted source. Trusted sources include other NWS, NOAA, or DOC, agencies, or other Federal agencies that operate at a FISMA high or equivalent level. Additionally, trusted sources could be third-party agencies through which there is an existing SLA on file (such as RedHat).
- [ ] The code is actively maintained and supported (it continues to get updates, patches, etc.)
- [ ] The code is not maintained by a private entity operating in a foreign country (if it is, make a note below)
- [ ] There is sufficient documentation to support maintenance
- [ ] There are no known security vulnerabilities or weaknesses
- [ ] Installing and running the code does not require privileged processes/users
- [ ] There are no software dependencies that are unapproved or have security concerns (if there are, make a note below)
- [ ] There are no concerns related to SA, SI, and SC NIST control families

### WCOSS2: Additional comments

_No response_

### LINKED ISSUE #1162: Turn off cylc variant for ewok-env by default, but make sure we test building cylc regularly
**Is your feature request related to a problem? Please describe.**
The cylc packages introduce a lot of Python dependencies that make concretization and building difficult and lengthy in the unified environment. Nobody so far is using the spack-stack cylc installation.

**Describe the solution you'd like**
NRL has an interest in using cylc from spack-stack, but doesn't need the full unified environment. We want to make sure that we continue testing the build of the cylc packages, so maybe it's possible to add it to the neptune environment (as a variant or as default). Or found another way (a chained environment tested in CI?).

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR adds the new py-wxflow package to global-workflow-env.

### Testing

tested concretization on hera

### Applications affected

None, now that jinja issue is resolved

### Systems affected

all

### Dependencies

none

### Issue(s) addressed

Fixes #1149
Fixes #1162

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji there's a hiccup here. py-cylc-flow wants `py-jinja2@3.0` (and no higher), but py-wxflow wants `py-jinja2@3.2.1:`. Looking at the py-cylc-flow repo, it looks like the jinja version limit is related to backward compatibility for older versions. Would it potentially work to relax the version requirement (and presumably forego that backward compatibility, whatever exactly that means)?

@aerorahul do you know where the jinja minimum required version for wxflow comes from? Is there any chance of relaxing it back to, say, 3.0.x?

**Comment by climbfuji** [comment]
> @climbfuji there's a hiccup here. py-cylc-flow wants `py-jinja2@3.0` (and no higher), but py-wxflow wants `py-jinja2@3.2.1:`. Looking at the py-cylc-flow repo, it looks like the jinja version limit is related to backward compatibility for older versions. Would it potentially work to relax the version requirement (and presumably forego that backward compatibility, whatever exactly that means)?
> 
> @aerorahul do you know where the jinja minimum required version for wxflow comes from? Is there any chance of relaxing it back to, say, 3.0.x?

I have no idea if relaxing the requirement for py-cylc-flow works. We are really stretching the limit with trying to get everything into one unified environment! Maybe it's time to reconsider. NOAA has no use for cylc at the moment, and conversely neither the Air Force nor JCSDA nor NRL care about py-wxflow.

**Comment by climbfuji** [comment]
That being said, take a look at #1162 - how about you turn the cylc variant off for ewok-env (in unified-dev and skylab-dev)? I can figure out if/how to add it elsewhere for testing.

File: configs/common/packages.yaml [hunk 1]
```diff
@@ -202,7 +202,7 @@
       require: '~mpi'
     # To avoid duplicate packages
     py-jinja2:
-      require: '@3.0.3'
+      require: '@3.1.2'
```
Comment (climbfuji) [inline configs/common/packages.yaml:205]: I think this is fine with neptune/py-xnrl. Let's merge it and see what happens.

**Comment by climbfuji** [comment]
Let me know when this is ready from your side, happy to merge later today

**Comment by AlexanderRichert-NOAA** [comment]
Yep good to go here.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**